keps/sig-node/127-user-namespaces/README.md

```diff
@@ -316,121 +325,121 @@ The picked range will be stored under a file named `userns` in the pod folder
-[atomic-writer-fsUser]: https://github.com/kubernetes/kubernetes/blob/cfd69463deeebfc5ae8a0813d7d2b125033c4aca/pkg/volume/util/atomic_writer.go#L68
-[configmap-fsuser]: https://github.com/kubernetes/kubernetes/blob/cfd69463deeebfc5ae8a0813d7d2b125033c4aca/pkg/volume/configmap/configmap.go#L272-L273
+To understand how idmap mounts work, let's first see an example of what happens
+without idmap mounts.
 
-#### Phase 2: pods with volumes
+If the pod wants to read who is the owner of file `/vol/configmap/foo`, as host
+UID 0 is not mapped, it will be shown as nobody. Therefore, it won't be able to
+read the file (unless it has permissions for others).
 
-This phase makes user namespaces work in pods with volumes too. This is
-activated via the bool `pod.spec.HostUsers` too and pods fall into this mode if
-some other volume type than the ones listed for phase 1 is used. IOW, when phase
-2 is implemented, pods that use volume types not supported in phase 1, fall into
-the phase 2.
+Therefore, if we want the pod to be able to read those files and permissions be
+what we expect, the kubelet needs to chown the file to UID 65536, as that is
+what root inside the container is mapped to in the user namespaces.
 
-All pods in this mode will use _the same_ mapping, chosen by the kubelet, with a
-length 65536, and mapping the range 0-65535 too. IOW, each pod will have its own user
-namespace, but they will map to _the same_ UIDs/GIDs in the host.
+We tried this before, but several limitations were hit:
 
-Using the same mapping allows for pods to share files and mitigates all the
-listed vulnerabilities (as the host is protected from the container). It is also
-a trivial next-step to take, given that we have phase 1 implemented: just return
-the same mapping if the pod has other volumes.
+* Changing the owner of files in configmaps/secrets/etc was a concern, those
+  volume type today ignore FsUser setting and changing it to honor them was a
+  concern for some members of the community due to possibly breaking some
+  existing behavior.
 
-While these pods do not use a distinct user namespace mapping, they are still
-using a new user namespace object in the kernel (so they cannot join/attack
-other pods namespaces). Security-wise this is a middle layer between what we
-have today (no userns at all) and using a distinct UID/GID mapping for the user
-namespace.
+* Therefore, we need to rely on `fsGroup`, that always mark files readable for
+  the group. This means this solution can't work when we NEED not to have
+  permissions for the group (think of ssh keys, for example, that tools enforce no
+  permission for the group)
 
-#### Phase 3: TBD
+* There are several other files that also need to have the proper permissions,
+  like: /dev/termination-log, /etc/hosts, /etc/resolv.conf, /etc/hotname, etc.
+  While it is completely possible to do, it adds more complexity to lot of parts
```

Comment:
I think adding some more details will be useful here to evaluate the challenges.

Reply:
Expanded with way more details and several links to specific examples in the code now. Please note this is in the alternatives section now